### DISCUSSION TITLE
Developers tutorials

### DISCUSSION THREAD

**Discussion by herve-gross** [GEOS-DEV/GEOS#1784] [Usability]
We need tutorials to help developers get started coding into GEOSX. Four topics have been identified so far:

1. Meshes and data on meshes: how to set and get mesh geometric information and the data living on various mesh-supported elements.
2. LAI: How to use the linear algebra interface.
3. Kernels: how to write good kernels.
4. Data structures: LVArrays, views, permutations and strides.

Please write down in this thread any **specific use case** you would like to see addressed in this context. the more specific, the better. For instance: how can we compute the volume of a given subset of elements, or how can we compute the average of cell-centered properties, etc.

We will prioritize these use cases based on their learning value and their usefulness and make PRs for each new tutorial.

**Comment by karimifard** [comment]
Here is a simple exercise in Mesh/Geometry category that can be very useful for a better understanding of GEOSX data structure.

1) Read an external unstructured mesh (hybrid).
2) Compute the volume of each element + memory allocation (using GEOSX data management) for the storage of the results.
3) Generate the list of adjacent elements (may require the use of temporary working memory).
4) Output specific subset of the model for visualization (selected elements).
